src/profile-logic/process-profile.js

```diff
@@ -1798,6 +1804,11 @@ export function processVisualMetrics(
 
     const changeMarkerName = `${metricName} Change`;
     for (const { timestamp, percent } of metric) {
+      if (timestamp === null) {
```

Comment:
Should we have the same check for the Progress markers above?

optional alternate change: put the check directly in `addMetricMarker` (and rename it `possiblyAddMetricMarker`) and remove from the loops.

Then I'd also remove the first condition `if (metric.every((m) => m.timestamp === null))`, indeed this will possibly loop over all metrics again in the case of the bug.

Reply:
I wasn't sure about adding the null checks to every addition at first because usually when the bug happens every timestamp becomes null, so I thought adding this check should fix it. But then couldn't help myself and added this just to be safe too :) But you're right, moving this to addMetricMarker is probably a safer option. Changed it.